File: backend/src/main/java/com/adora/entity/Report.java
package com.adora.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "reports")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Report {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "reporter_id", nullable = false)
    private User reporter;

    @Column(name = "target_type", nullable = false)
    private String targetType; // e.g. "BILLBOARD", "USER"

    @Column(name = "target_id", nullable = false)
    private Long targetId;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String reason;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ReportStatus status;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
}

File: src/types/report.ts
import { User } from './user';

export type ReportStatus = 'PENDING' | 'RESOLVED' | 'REJECTED';

export interface ReportDto {
  id: number;
  reporter?: User;
  targetType: string;
  targetId: number;
  reason: string;
  status: ReportStatus;
  createdAt: string;
}

export interface CreateReportRequest {
  targetType: 'BILLBOARD' | 'USER';
  targetId: number;
  reason: string;
}
